lib.php

```diff
@@ -529,4 +529,62 @@ function (stdClass $warning) use ($userid) : bool {
     foreach ($updatedwarnings as $updatedwarning) {
         $DB->update_record('attendance_warning', $updatedwarning);
     }
+}
+
+/**
+ * This function receives a calendar event and returns the action associated with it.
+ *
+ * This is used by block_timeline in order to display the event appropriately. If null is returned then the event
+ * is not displayed on the block.
+ *
+ * @param calendar_event $event
+ * @param \core_calendar\action_factory $factory
+ * @param int $userid
+ * @return \core_calendar\local\event\entities\action_interface|\core_calendar\local\event\value_objects\action
+ */
+function mod_attendance_core_calendar_provide_event_action(calendar_event $event,
```

Comment:
The implementation of this function appears to be a bit different from core which uses modinfo and checks uservisible and makes capability checks before doing any sql queries - I also see some use in core functions of global $USER when userid = 0

ideally we should be writing unit tests for this change too - it appears that this stuff is quite easy to test and there are a number of examples of unit tests for existing plugins that could be used